tests/api.test.js

```diff
@@ -2,123 +2,108 @@ import { describe } from "node:test";
 import { test, expect, request } from "vitest";
 import request from "supertest";
 
-const URL = "https://app.swaggerhub.com/apis/GREENE8616/BeKindy/1.0.0/";
+const URL = "http://localhost:5000";
 
-describe("USERS table tests", () => {
-  test("GET from /api/v1/users", async () => {
-    const res = await request(URL).get("/api/v1/users");
+describe("USERS SUITE", () => {
+  test("should get users and return status code 200", async () => {
+    const res = await request(URL).get("/api/v1/users/");
     expect(res.statusCode).toBe(200);
   });
+});
 
-  test("POST to /api/v1/users", async () => {
-    const res = await request(URL)
-      .post("/api/v1/users")
-      .send({ username: "JohnDoe01", email: "JohnDoe@example.mail" });
-    expect(res.statusCode).toBe(201);
-    expect(res.body.username).toBe("JohnDoe01");
-    expect(res.body.email).toBe("JohnDoe@example.mail");
+describe("CHALLENGES SUITE", () => {
+  test("should get today's challenge and return status code 200", async () => {
+    const res = await request(URL).get("/api/v1/challenges/today");
+    expect(res.statusCode).toBe(200);
   });
 
-  test("PUT to /api/v1/users/1", async () => {
-    const res = await request(URL)
-      .put("/api/v1/users/1")
-      .send({ name: "Jane Doe", age: 30 });
-    expect(res.statusCode).toBe(200);
-    expect(res.body.name).toBe("JaneDoe");
-    expect(res.body.email).toBe("something@example.mail");
+  test("should try to get challenge with incorrect date and return status code 404", async () => {
+    const res = await request(URL).get("/api/v1/challenges/2021-01-01");
+    expect(res.statusCode).toBe(404);
   });
 
-  // test("DELETE /api/v1/users/1", async () => {
-  //   const res = await request(URL).delete("/api/v1/users/1");
-  //   expect(res.statusCode).toBe(204);
-  // });
+  test("should send bad request and return status code 400", async () => {
+    const res = await request(URL).get("/api/v1/challenges/foobar/");
+    expect(res.statusCode).toBe(400);
+  });
+
+  test("should request nonexistent challenge and return status code 404", async () => {
+    const res = await request(URL).get("/api/v1/challenges/1000");
+    expect(res.statusCode).toBe(404);
+  });
```

Comment:
This check works with our current data set (and we will never have more than 1000 challenges) but in "the real world" this test case could fail because of our data set, not because of bad code.

Reply:
Yeah. The idea is just to have a number significantly greater than the user count to just make sure it'll return that error. But ideally I probably wouldn't hardcode 1000